Gutenboarding: add new template designs

ramonjd opened the pull request:
## Changes proposed in this Pull Request

⚠️ https://github.com/Automattic/wp-calypso/pull/42353 ~must be merged first~ merged

## Questions

1. What is the relationship between theme and design/template? For example, how do we distinguish between [Bowen](https://gutenboardinghp1.wordpress.com/) and [Coutoire](https://gutenboardinghp20.wordpress.com/)? 
2. I've been determining the fonts by looking at the heading and base fonts for the demo sites, e.g., https://coutoiredemo.wordpress.com/ Is this correct?

## TODO
- [x] Add the rest of the new themes/designs from #42353
- [ ] Get some indication on how to properly test this 
- [x] Add feature flag for testing? https://github.com/Automattic/wp-calypso/pull/42353#issuecomment-630672970
- [ ] Ask if any of the new designs should be "premium". See: https://github.com/Automattic/wp-calypso/pull/42377
- [x] Generate preview images: https://github.com/Automattic/wp-calypso/blob/master/bin/generate-gutenboarding-design-thumbnails.js (Re-generate after https://github.com/Automattic/wp-calypso/pull/43104 lands)

## Testing instructions

Head over to /new
Make sure the following designs feature in /new/design: [TO COME]
Ensure that the font selector and preview works for each design
Create a site using each design, checking that the resulting homepage matches the theme/design

Fixes #42208

ramonjd commented on static/page-templates/page-templates.json:
```diff
@@ -89,6 +94,10 @@
+		},
+		{
+			"title": "barnsbury",
+			"demo": "https://barnsburydemo.wordpress.com/"
```
What the point of this array? Does the `public-api.wordpress.com/rest/v1/template/demo` API need it to generate the demo?

andrewserong replied:
Apologies for the 28-days-late answer to this question, but that API endpoint doesn't use this file. I don't _think_ the file is currently used anywhere at all, actually! I wonder if this file was a precursor to what we now have in `available-designs-config.json`? For consistency though, I don't see an issue with adding them here, and eventually we'll want to have an endpoint that can list info similar to this, I suppose...

ramonjd replied:
> I don't think the file is currently used anywhere at all, actually! 

Thanks! That's what I was thinking! I just wanted to make sure I wasn't missing something again. 🥴

simison replied:
I don't think it is used anywhere neither.

@johnHackworth remember details?

ramonjd replied:
I just came across it yesterday. It's used by the verticals MC tool only. :)

There might be scope to simplify/merge these config files. We're moving them over to the server for now, mainly to reduce the number of remote gets we're doing in our APIs, but also since devs are the only ones updating them anyway.

simison replied:
> There might be scope to simplify/merge these config files.

Yep, see https://github.com/Automattic/wp-calypso/pull/40587

ramonjd commented on client/landing/gutenboarding/available-designs.ts:
```diff
@@ -120,6 +120,18 @@ const availableDesigns: Readonly< AvailableDesigns > = {
+			slug: 'barnsbury',
+			template: 'barnsbury',
+			theme: 'barnsbury',
+			src: 'https://public-api.wordpress.com/rest/v1/template/demo/barnsbury/barnsbury/',
```
@simison I can't remember how you generated the static images, e.g., https://github.com/Automattic/wp-calypso/pull/40868 

I just took a screenshot from the demo file in https://github.com/Automattic/wp-calypso/pull/42353

ramonjd replied:
Update: just saw that you replied in slack 🙇 

Thanks!!!

> Instructions on top of https://github.com/Automattic/wp-calypso/blob/master/bin/generate-gutenboarding-design-thumbnails.js

ianstewart commented:
> I've been determining the fonts by looking at the heading and base fonts for the demo sites, e.g., https://coutoiredemo.wordpress.com/ Is this correct?

@alaczek has prepared a set of source sites that use fonts available in the Gutenboarding flow for these "starter designs".

- [Barnsbury](https://gutenboardinghp11.wordpress.com/)
- [Mayland](https://gutenboardinghp12.wordpress.com/)
- [Rivington](https://gutenboardinghp13.wordpress.com/)
- [Maywood](https://gutenboardinghp14.wordpress.com/)
- [Balasana](https://gutenboardinghp15.wordpress.com/)
- [Alves](https://gutenboardinghp16.wordpress.com/)
- [Stratford](https://gutenboardinghp17.wordpress.com/)
- [Rockfield](https://gutenboardinghp18.wordpress.com/)

@alaczek can you work with Ramon's team to ensure these display as intended?

ramonjd commented:
> @alaczek has prepared a set of source sites that use fonts available in the Gutenboarding flow for these "starter designs".

There were a couple of source sites in @ianstewart's spreadsheet I wasn't sure about. Here's what I could glean from them. 

**Coutoir**
https://gutenboardinghp20.wordpress.com/

_headings_: `'EB Garamond'`,
_base_: `'Work Sans',`

**Leven**
https://gutenboardinghp21.wordpress.com/
_headings_:  `-apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;`

_base_: `-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;`

They're using fonts we're not loading in Gutenboarding at the moment. Would you know which we should use (if any)?

Thanks

alaczek commented:
There were two sets mentioned in #42120. I only did the [first set ](https://github.com/Automattic/wp-calypso/issues/42120#issuecomment-628075015).

Twenty Twenty, Coutoire, and Leven are not updated, so they should not be included in this PR. But I'll add this to my to-do list and try to tackle tomorrow or on Monday. So you could either:
* wait for me to update the second set, then update this PR and deploy
* remove them from this PR, deploy, and start a new PR when the second set is ready.

ramonjd commented:
> Twenty Twenty, Coutoire, and Leven are not updated, so they should not be included in this PR. But I'll add this to my to-do list and try to tackle tomorrow or on Monday. 

This PR has been slow moving, so I dare say we'll give you plenty of time to work your magic on the other designs, unless you feel strongly either way.

Also happy to get the other designs out there for testing. They're behind a feature flag and won't go to production anyway.

 Thank you @alaczek !!!

alaczek commented:
I did the second set, awaiting feedback from @ianstewart, and then we can move forward.

alaczek commented:
Alrighty, I updated Coutoire and Leven. 

Coutoire: Playfair display + Fira Sans

Leven: Chivo + Open Sans, plus I removed a bunch of content from the homepage.

They are ready to go so you can update this PR and get it moving again. We're dropping Twenty Twenty from the lineup though. Let me know if you need me to test/review anything.

ramonjd commented:
> They are ready to go so you can update this PR and get it moving again. We're dropping Twenty Twenty from the lineup though. Let me know if you need me to test/review anything.

Thanks @alaczek !! I've updated the fonts and regenerated the screenshots. Also removed Twenty Twenty

Take a look at `/new` and let me know if you see anything strange.

Still not sure if any of the new designs should be "premium". You'll see some of them have premium badges on the design step.

andrewserong commented on config/stage.json:
```diff
@@ -46,6 +46,7 @@
 		"google-my-business": true,
 		"gutenboarding/mshot-preview": false,
 		"gutenboarding/style-preview-verticals": false,
+		"gutenboarding/edge-templates": true,
```
Do we need to have Staging show the edge templates as well? Just thinking that for debugging something happening in Production, it might be easier for us if Staging is as close as possible to what the user is seeing in Production, and we could use horizon or wpcalypso for testing Edge templates?

ramonjd replied:
> Do we need to have Staging show the edge templates as well? 

No we absolutely don't. Thanks a lot for catching this. It was a shotgun approach. I'll fix now.

andrewserong commented:
@ramonjd these are all working well, the only issue I found is that the screenshot for Rockfield is missing a background image behind "Discover our Menu":

![image](https://user-images.githubusercontent.com/14988353/84721530-3ec88180-afc4-11ea-8002-a2e45720a4da.png)

But otherwise, tested and created sites using the following designs, and all looked good to me!

* Barnsbury - ✅ 
* Alves - ✅ 
* Mayland - ✅ 
* Rivington - ✅ 
* Maywood - ✅ 
* Balasana - ✅ 
* Stratford - ✅ 
* Rockfield - the screenshot needs to be updated, but otherwise ✅ 
* Coutoire - ✅ 
* Leven - ✅ 
* Gibbs - ✅

ramonjd commented:
> Rockfield - the screenshot needs to be updated, but otherwise

Thanks for the thorough test!!! I had to generate that one manually because the screenshot script kept generated a huge picture of a hand only 🤣 

I'll try again 🙇

alaczek commented:
Looking good aside from a few minor issues.

![image](https://user-images.githubusercontent.com/4550351/84727833-4a6f7480-afd3-11ea-88a2-fcec374e9dfd.png)
Two things in this screenshot:
* the site title looks off in the preview, it should be bold (this is Barnsbury, but the same happens in all themes that have a bold site title)
* do we want to display the inverse font pairings? This feels redundant. 

And lastly, in Cutuoire the body font in preview is Work Sans, should be Fira Sans.

andrewserong commented:
> the site title looks off in the preview, it should be bold (this is Barnsbury, but the same happens in all themes that have a bold site title)

Just thought I'd take a quick look at this. I think the issue here might be that the `template/demo` endpoint appears to strip the HTML elements within the `.site-title` element when swapping out the text there for the site title, so it doesn't have the wrapping `a` tag to give it the bold style via `.site-title a`. Maybe something for us to fix up in that endpoint separately?

ramonjd commented:
Thanks @alaczek 

> do we want to display the inverse font pairings? This feels redundant.

It's a good question. It seems that we [display all possible "heading/base" combinations](https://github.com/Automattic/wp-calypso/blob/734d1e6/client/landing/gutenboarding/onboarding-block/style-preview/font-select.tsx#L86).

So we'll show each of these:

[Stratford](https://github.com/Automattic/wp-calypso/pull/42354/files#diff-7fa2a963ce8fa39f90583b3b1708b3a3R218)

```
				"headings": "Chivo",
				"base": "Open Sans"
```


[Balasana](https://github.com/Automattic/wp-calypso/pull/42354/files#diff-7fa2a963ce8fa39f90583b3b1708b3a3R204)

```
				"headings": "Open Sans",
				"base": "Chivo"
```

Unless I got the order wrong and it's a mistake?

Otherwise I think it makes sense to keep the inverse pair no matter how strange it looks. For example, if we remove Open Sans/Chivo but keep Chivo/Open Sans, a user who selects Balasana won't be able to switch back to the original font setting if they change it.

> the site title looks off in the preview, it should be bold (this is Barnsbury, but the same happens in all themes that have a bold site title)

This is current behaviour it seems. I'll file a separate issue.

ramonjd commented:
> the site title looks off in the preview, it should be bold (this is Barnsbury, but the same happens in all themes that have a bold site title)

Tracking here: https://github.com/Automattic/wp-calypso/issues/43310

alaczek commented:
> Unless I got the order wrong and it's a mistake?

Nope, you got the order right - the font pairings can be reversed.

> Otherwise I think it makes sense to keep the inverse pair no matter how strange it looks. For example, if we remove Open Sans/Chivo but keep Chivo/Open Sans, a user who selects Balasana won't be able to switch back to the original font setting if they change it.

I don't quite understand. Why would they not be able to switch back? 

I'm not sure what the best solution is here tbh, but I imagined the font pairings could be changing based on the current layout. So for example, if someone picks Balasana, we offer Open Sans / Chivo, and remove the inverse pair. This is sort of happening already for Stratford - it is using Chivo / Open Sans and not offering the inverse pair.

cc @ianstewart and @dubielzyk for thoughts on this one.

ramonjd commented:
> So for example, if someone picks Balasana, we offer Open Sans / Chivo, and remove the inverse pair. 

Ah, sorry. I think I understand what you mean now. 

So we should always show the default font combination, and remove the inverse. So the user can, for example, go with Open Sans / Chivo for Balasana, but never Chivo / Open Sans .

This is something we could do. It's probably best suited for another PR as it changes existing functionality.

> This is sort of happening already for Stratford - it is using Chivo / Open Sans and not offering the inverse pair.

This is by chance, and not design I think. Since Balasana introduces a new pair combo [not yet known by our default list](https://github.com/Automattic/wp-calypso/blob/734d1e6fe9e716543b0766dc85e7a7bf9f71717e/client/landing/gutenboarding/constants.ts#L37), it adds it to the top of the button list.

noahtallen commented:
it'd be super nice to at least get the `is_edge` option merged! (that's the only part I'm blocked on at the moment.) Is there anything I can do to test or unblock that?

Thinking about the name, it could be confused with "gutenberg edge". just throwing some alternative ideas out:

- `is_alpha`
- `is_beta`
- `is_dev`
- `is_staging`
- `restrict_to_staging` (or replace staging with alpha/beta/dev)

ramonjd commented:
@alaczek I'm seeing something odd with [Rivington](https://gutenboardinghp13.wordpress.com/).

The mid section text is no longer dark. I can't remember when it was, but I've never noticed the light text on white before in my testing.

<img width="803" alt="Screen Shot 2020-06-18 at 11 19 53 am" src="https://user-images.githubusercontent.com/6458278/84966776-b5e14f80-b155-11ea-9a7c-f7db7efbb3e4.png">

It's the same on the demo site: https://rivingtondemo.wordpress.com/

ramonjd commented:
> it'd be super nice to at least get the is_edge option merged! (that's the only part I'm blocked on at the moment.) Is there anything I can do to test or unblock that?

I like `is_alpha`

I'll change it to that and then I can probably merge this PR. 

It won't affect stage or production and so we can deal with design issues in another PR.

noahtallen commented:
> so we can deal with design issues in another PR

Nice! I guess that's the point of having alpha designs anyways.

ramonjd commented:
Following up this PR's open questions/tasks in another PR: https://github.com/Automattic/wp-calypso/issues/43420

ianstewart commented:
@allancole this looks like it's from the Rivington colour scheme that just shipped, I think.

kjellr commented:
> this looks like it's from the Rivington colour scheme that just shipped, I think.

This should be fixed as of https://github.com/Automattic/themes/pull/2153. 👍
